Sources/Hummingbird/Middleware/FileMiddleware.swift

```diff
@@ -95,7 +102,7 @@ public struct FileMiddleware<Context: RequestContext, Provider: FileProvider>: R
             }
 
             // Remove percent encoding from URI path
-            guard let path = request.uri.path.removingPercentEncoding else {
+            guard var path = request.uri.path.removingPercentEncoding else {
```

Comment:
I think before we do anything, we should actually check if the path start with the pathPrefix.
this way can mount several FileMiddlewares on different URL paths (eg: each service a separate file system directoy)

Reply:
also, a quick-bail if the method is not `get` or `head` would be a good idea I think

Reply:
I added the quick bail for get and head. Also added bailing on path prefix not existing. But this needs to be done after the percent encoding is removed. Test is updated to include 2 middleware which return slightly different results